From 39b579459fd42703731f833057d49ef120400c5f Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 11 May 2023 15:44:18 +0100
Subject: [PATCH] ARM: dts: bcm2709/10: Retain the system-timer node

Although the system timer is largely ignored in favour of the ARM local
timers, retain the DT node so that the bcm2835-sdhost logging can find
the timer in a cleaner fashion.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2709.dtsi | 6 ++++--
 arch/arm/boot/dts/bcm2710.dtsi | 6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

--- a/arch/arm/boot/dts/bcm2709.dtsi
+++ b/arch/arm/boot/dts/bcm2709.dtsi
@@ -10,8 +10,6 @@
 
 		dma-ranges = <0xc0000000 0x00000000 0x3f000000>,
 			     <0x7e000000 0x3f000000 0x01000000>;
-
-		/delete-node/ timer@7e003000;
 	};
 
 	__overrides__ {
@@ -22,6 +20,10 @@
 	};
 };
 
+&system_timer {
+	status = "disabled";
+};
+
 &vc4 {
 	status = "disabled";
 };
--- a/arch/arm/boot/dts/bcm2710.dtsi
+++ b/arch/arm/boot/dts/bcm2710.dtsi
@@ -13,8 +13,6 @@
 	soc {
 		dma-ranges = <0xc0000000 0x00000000 0x3f000000>,
 			     <0x7e000000 0x3f000000 0x01000000>;
-
-		/delete-node/ timer@7e003000;
 	};
 
 	__overrides__ {
@@ -25,6 +23,10 @@
 	};
 };
 
+&system_timer {
+	status = "disabled";
+};
+
 &vc4 {
 	status = "disabled";
 };
